.github/workflows/reusable-on-code-change.yml

```diff
@@ -0,0 +1,198 @@
+name: Reusable Workflow (On Code Change)
```

Comment:
Having a single reusable workflow for on-code-change and on-merge might be a good idea as you suggest. if that's the case, you might want to rename this.